Tests/SwiftSyntaxTest/SyntaxVisitorTests.swift

```diff
@@ -0,0 +1,248 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import SwiftSyntax
+import SwiftParser
+import SwiftSyntaxBuilder
+import _SwiftSyntaxTestSupport
+
+extension Parser {
+  public static func parse(_ url: URL) -> SourceFileSyntax {
+    let source = try! String(contentsOf: url)
+    return Self.parse(source: source)
+  }
+}
+
+public class SyntaxVisitorTests: XCTestCase {
+  /// The syntax tree of the following source file
+  /// ```swift
+  /// func foo() {
+  ///   public func foo() {
+  ///     func foo() {
+  ///       /*Unknown token */0xG
+  ///     }
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// The source file is hard-coded so this test case doesn't need to depend on the parser.
```

Comment:
Is there some reason not to depend on the parser here?

Reply:
It’s just that this test is in `SwiftSyntaxTest` and I don’t think we should be depending on the parser from `SwiftSyntaxTest` because it’s lower level. Also, it didn’t make any sense for me to move this file to `SwiftParserTest` because that’s not where it belongs, so I ended up with this solution, which I don’t find super terrible.